DEVELOPMENT.md

```diff
@@ -2,168 +2,255 @@
-
--   Update a single unit test golden file, for example `JavaCodeGeneratorTest.java`:
-
-    ```sh
-    bazel run //:update_com_google_api_generator_engine_JavaCodeGeneratorTest
-    ```
-    or
-    ```sh
-    mvn test -DupdateUnitGoldens -Dtest=JavaCodeGeneratorTest
-    ```
-
--   Run a single integration test for API like `Redis`, it generates Java source
-    code using the Java microgenerator and compares them with the goldens files
-    in `test/integration/goldens/redis`.
-
-    ```sh
-    bazel test //test/integration:redis
-    ```
-
--   Update integration test golden files, for example `Redis`. This clobbers all the
-    files in `test/integration/goldens/redis`.
-
-    ```sh
-    bazel run //test/integration:update_redis
-    ```
+  ```sh
+  bazel test //:units
+  ```
+  or
+  ```sh
+  mvn test
+  ```
+
+- Run a single unit test like `JavaCodeGeneratorTest.java`:
+
+  ```sh
+  bazel test //:unit_com_google_api_generator_engine_JavaCodeGeneratorTest
+  ```
+  or
+  ```sh
+  mvn test -Dtest=JavaCodeGeneratorTest
+
+  mvn test "-Dtest=Basic*, !%regex[.*.Unstable.*], !%regex[.*.MyTest.class#one.*|two.*], %regex[#fast.*|slow.*]"
+  ```
+
+- Update all unit test golden files:
+
+  ```sh
+  mvn test -DupdateUnitGoldens
+  ```
+
+- Update a single unit test golden file, for example `JavaCodeGeneratorTest.java`:
+
+  ```sh
+  bazel run //:update_com_google_api_generator_engine_JavaCodeGeneratorTest
+  ```
+  or
+  ```sh
+  mvn test -DupdateUnitGoldens -Dtest=JavaCodeGeneratorTest
+  ```
+
+## Golden Integration Testing
+
+- Run a single integration test for API like `Redis`, it generates Java source
+  code using the Java microgenerator and compares them with the goldens files
+  in `test/integration/goldens/redis`.
+
+  ```sh
+  bazel test //test/integration:redis
+  ```
+
+- Update integration test golden files, for example `Redis`. This clobbers all the
+  files in `test/integration/goldens/redis`.
+
+  ```sh
+  bazel run //test/integration:update_redis
+  ```
+
+### Golden Showcase Testing
+
+- Regenerate the showcase client into the `./showcase` projects.
+
+  ```shell
+  mvn compile -P update-showcase
+  # or, for backward compatibility with previous workflows:
+  mvn compile -D updateUnitGoldens
+  ```
+
+- Run a test to verify a newly generated showcase client is identical to the golden files
+  in `./showcase`.
+
+  ```shell
+  mvn verify -P showcase-golden
+  ```
+
+## Showcase Integration Testing
+
+[GAPIC Showcase](https://github.com/googleapis/gapic-showcase) is an API that demonstrates Generated
+API Client (GAPIC) features and common API patterns used by Google. It follows the [Cloud APIs
+design guide](https://cloud.google.com/apis/design/). `gapic-generator-java` generates a client for
+the Showcase API which can communicate with a local Showcase server to perform integration tests.
+
+### Requirements
+
+* Install [Go](https://go.dev) in your `PATH`.
+
+### Installing the Server
+
+```shell
+$ GAPIC_SHOWCASE_VERSION=0.25.0
```

Comment:
Can we mention that the showcase version should be the latest from `gapic-showcase` repo?

Reply:
- [ ] Update automatically with renovate bot
- [ ] Add documentation to use the latest server version

Reply:
Based on our in-person discussion, I'm resolving this as a "nice to have but not required" feature.